[2025-08-18T09:56:37.307+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-08-18T09:56:37.347+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: youtube_pipeline_simple.golden_to_star manual__2025-08-18T09:36:47.534001+00:00 [queued]>
[2025-08-18T09:56:37.357+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: youtube_pipeline_simple.golden_to_star manual__2025-08-18T09:36:47.534001+00:00 [queued]>
[2025-08-18T09:56:37.358+0000] {taskinstance.py:2303} INFO - Starting attempt 2 of 2
[2025-08-18T09:56:37.372+0000] {taskinstance.py:2327} INFO - Executing <Task(BashOperator): golden_to_star> on 2025-08-18 09:36:47.534001+00:00
[2025-08-18T09:56:37.378+0000] {standard_task_runner.py:63} INFO - Started process 2441 to run task
[2025-08-18T09:56:37.382+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'youtube_pipeline_simple', 'golden_to_star', 'manual__2025-08-18T09:36:47.534001+00:00', '--job-id', '43', '--raw', '--subdir', 'DAGS_FOLDER/youtube_pipeline.py', '--cfg-path', '/tmp/tmplmvs86ub']
[2025-08-18T09:56:37.384+0000] {standard_task_runner.py:91} INFO - Job 43: Subtask golden_to_star
[2025-08-18T09:56:37.446+0000] {task_command.py:426} INFO - Running <TaskInstance: youtube_pipeline_simple.golden_to_star manual__2025-08-18T09:36:47.534001+00:00 [running]> on host 4d75244235f4
[2025-08-18T09:56:37.541+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='youtube_pipeline_simple' AIRFLOW_CTX_TASK_ID='golden_to_star' AIRFLOW_CTX_EXECUTION_DATE='2025-08-18T09:36:47.534001+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-18T09:36:47.534001+00:00'
[2025-08-18T09:56:37.542+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-08-18T09:56:37.566+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-08-18T09:56:37.567+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'docker exec youtube-pipeline-project_spark-master_1 spark-submit --master spark://spark-master:7077 --deploy-mode client --packages org.postgresql:postgresql:42.6.0 /opt/***/jobs/star_schema/golden_to_star_citus.py']
[2025-08-18T09:56:37.578+0000] {subprocess.py:86} INFO - Output:
[2025-08-18T09:56:39.834+0000] {subprocess.py:93} INFO - :: loading settings :: url = jar:file:/opt/bitnami/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-08-18T09:56:39.912+0000] {subprocess.py:93} INFO - Ivy Default Cache set to: /opt/bitnami/spark/.ivy2/cache
[2025-08-18T09:56:39.913+0000] {subprocess.py:93} INFO - The jars for the packages stored in: /opt/bitnami/spark/.ivy2/jars
[2025-08-18T09:56:39.918+0000] {subprocess.py:93} INFO - org.postgresql#postgresql added as a dependency
[2025-08-18T09:56:39.919+0000] {subprocess.py:93} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-e643cfe7-8560-4602-9cb6-3394177f060d;1.0
[2025-08-18T09:56:39.920+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-08-18T09:56:40.085+0000] {subprocess.py:93} INFO - 	found org.postgresql#postgresql;42.6.0 in central
[2025-08-18T09:56:40.119+0000] {subprocess.py:93} INFO - 	found org.checkerframework#checker-qual;3.31.0 in central
[2025-08-18T09:56:40.162+0000] {subprocess.py:93} INFO - :: resolution report :: resolve 224ms :: artifacts dl 18ms
[2025-08-18T09:56:40.163+0000] {subprocess.py:93} INFO - 	:: modules in use:
[2025-08-18T09:56:40.164+0000] {subprocess.py:93} INFO - 	org.checkerframework#checker-qual;3.31.0 from central in [default]
[2025-08-18T09:56:40.165+0000] {subprocess.py:93} INFO - 	org.postgresql#postgresql;42.6.0 from central in [default]
[2025-08-18T09:56:40.165+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-08-18T09:56:40.165+0000] {subprocess.py:93} INFO - 	|                  |            modules            ||   artifacts   |
[2025-08-18T09:56:40.166+0000] {subprocess.py:93} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-08-18T09:56:40.167+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-08-18T09:56:40.167+0000] {subprocess.py:93} INFO - 	|      default     |   2   |   0   |   0   |   0   ||   2   |   0   |
[2025-08-18T09:56:40.171+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-08-18T09:56:40.172+0000] {subprocess.py:93} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-e643cfe7-8560-4602-9cb6-3394177f060d
[2025-08-18T09:56:40.173+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-08-18T09:56:40.186+0000] {subprocess.py:93} INFO - 	0 artifacts copied, 2 already retrieved (0kB/13ms)
[2025-08-18T09:56:40.446+0000] {subprocess.py:93} INFO - 25/08/18 09:56:40 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-08-18T09:56:41.513+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-08-18T09:56:41.514+0000] {subprocess.py:93} INFO -   File "/opt/***/jobs/star_schema/golden_to_star_citus.py", line 2, in <module>
[2025-08-18T09:56:41.514+0000] {subprocess.py:93} INFO -     import psycopg2 # type: ignore
[2025-08-18T09:56:41.514+0000] {subprocess.py:93} INFO -     ^^^^^^^^^^^^^^^
[2025-08-18T09:56:41.515+0000] {subprocess.py:93} INFO - ModuleNotFoundError: No module named 'psycopg2'
[2025-08-18T09:56:41.566+0000] {subprocess.py:93} INFO - 25/08/18 09:56:41 INFO ShutdownHookManager: Shutdown hook called
[2025-08-18T09:56:41.568+0000] {subprocess.py:93} INFO - 25/08/18 09:56:41 INFO ShutdownHookManager: Deleting directory /tmp/spark-6d3a3761-9084-4d5f-ae3e-ca39d0fc2cd3
[2025-08-18T09:56:41.611+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-08-18T09:56:41.612+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-08-18T09:56:41.625+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/bash.py", line 243, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-08-18T09:56:41.630+0000] {taskinstance.py:1205} INFO - Marking task as FAILED. dag_id=youtube_pipeline_simple, task_id=golden_to_star, execution_date=20250818T093647, start_date=20250818T095637, end_date=20250818T095641
[2025-08-18T09:56:41.645+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 43 for task golden_to_star (Bash command failed. The command returned a non-zero exit code 1.; 2441)
[2025-08-18T09:56:41.658+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-08-18T09:56:41.677+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-08-18T09:56:41.679+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
